### PR TITLE
Improve hdbcli types

### PR DESCRIPTION
**Pull request by kasium**
* parameters argument of execute methods is optional and allows none values
* paramstyle is a str and a tuple of str (hdbcli does not strictly follow PEP 249)
* Add missing functions to ResultRow

### CODE REVIEW

File: stubs/hdbcli/hdbcli/dbapi.pyi [hunk 1]
```diff
@@ -4,13 +4,13 @@ from collections.abc import Sequence
...
 from typing import Any, overload
-from typing_extensions import Literal, TypeAlias
+from typing_extensions import Literal, Self, TypeAlias
 
 from .resultrow import ResultRow
 
 apilevel: str
 threadsafety: int
-paramstyle: tuple[str, ...]
+paramstyle: str | tuple[str, ...]  # hdbcli defines it as a tuple which does not follow PEP 249
```
Comment (srittau) [inline stubs/hdbcli/hdbcli/dbapi.pyi:13]: I like the comment, but when it's defined as a tuple, why the `str` alternative?

Comment (kasium) [reply stubs/hdbcli/hdbcli/dbapi.pyi:13]: Valid point. I thought that there is a use case setting a value, but there is not

**Comment by kasium** [comment]
@srittau can you please consider merging it? Thanks a lot!

**Comment by srittau** [comment]
@kasium See my last comment: Could you re-add the comment about paramstyle being non-standard?

**Review by srittau — APPROVED** [review]

**Pull request merged**